### ISSUE TITLE
Enum validation fails for numbers

### ISSUE THREAD

**Issue by ksheedlo** [mozilla-services/react-jsonschema-form#1098]
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.

### Description

react-jsonschema-form fails validation when the schema field has `enum` validation and a number is selected.

### Steps to Reproduce

1. Go to https://codesandbox.io/s/21j3m9m5qr
2. Select 1 from the dropdown (also works with 2 or 3)
3. Click submit

#### Expected behavior

The demo application should display the following content JSON in the pre tag:

```
{
  "count": 1
}
```

#### Actual behavior

Validation failed, and the following error message was produced.

<img width="620" alt="screen shot 2018-12-05 at 4 24 15 pm" src="https://user-images.githubusercontent.com/1271410/49552903-4aed6200-f8aa-11e8-8349-b36272c92f44.png">

### Version

1.0.6

**Comment by LucianBuzzo** [comment]
The issue here is that the default `Select` widget just uses `event.target.value` to retrieve the selected value on change, which by default is always a string. https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/widgets/SelectWidget.js#L32

**Comment by LucianBuzzo** [comment]
You can work around this by setting the `type` keyword in your schema, so that `react-jsonschema-form` can correctly infer the type you are expecting, for example:
```
const SCHEMA = {
  type: "object",
  properties: {
    count: {
      enum: [1, 2, 3],
      type: "number"
    }
  }
};
```

**Comment by LucianBuzzo** [comment]
I've opened a PR to infer the type from the `enum` if `type` is not present in the schema https://github.com/mozilla-services/react-jsonschema-form/pull/1100

**Comment by glasserc** [comment]
As #1100 was merged and released in v1.0.7, I'm closing this. Please let me know if the problem persists.
